src/auto_pickup.cpp

```diff
@@ -32,7 +32,10 @@ void auto_pickup::show()
 
 void auto_pickup::show( const std::string &custom_name, bool is_autopickup )
 {
-    save_reset_changes(false);
+    std::vector<cRules> vRulesOld[MAX_TAB];
```

Comment:
If you change `vRules` to a `std::array<std::vector<...>, MAX_TAB>`, you can reduce this to one line: `decltype(vRules) vRulesOld = vRules;` - the `std::array` has the appropriate copy constructor that takes care of all elements.

It also allows range-based for loops, useful when you don't actually need the index.